compiler/src/iree/compiler/Dialect/Util/IR/UtilAttrs.cpp

```diff
@@ -423,6 +435,56 @@ static LogicalResult serializeGenericElementData(Location loc,
   return emitError(loc) << "unhandled constant type " << elementsAttr.getType();
 }
 
+// Performs serialization of all of the elements in |resourceElementsAttr|.
+// Throws error if not supported.
+static LogicalResult serializeGenericResourceElementData(
+    Location loc, DenseResourceElementsAttr resourceElementsAttr,
+    llvm::endianness endian, llvm::raw_ostream &os) {
+
+  if (endian != llvm::endianness::native) {
+    return emitError(loc) << "the endian of the "
+                             "DenseResourceElementsAttr is not supported";
+  }
+  if (llvm::isa<IntegerType>(resourceElementsAttr.getType().getElementType())) {
+    // Don't hoist bitWidth given `getElementTypeBitWidth()` asserts if the
```

Comment:
Can you add a note for posterity:

// At the time of writing, DenseResourceElementsAttr byte aligned physical element types only
// with the exception of `i1`, which is stored as a full byte. This is in contrast to DenseElementsAttr
// which has an exception for `i1` where it is bit-packed.

Reply:
yeah, done